File: app/src/main/java/com/drnoob/datamonitor/ui/fragments/ContributorsFragment.java
/*
 * Copyright (C) 2021 Dr.NooB
 *
 * This file is a part of Data Monitor <https://github.com/itsdrnoob/DataMonitor>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.drnoob.datamonitor.ui.fragments;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.preference.PreferenceFragmentCompat;

import com.drnoob.datamonitor.R;

import org.jetbrains.annotations.NotNull;

public class ContributorsFragment extends Fragment {
    private TextView mContribute;

    public ContributorsFragment() {
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull @NotNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_contributors, container, false);

        mContribute = view.findViewById(R.id.contribute);

        mContribute.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(Intent.ACTION_VIEW).setData(Uri.parse(getString(R.string.github_contribute))));
            }
        });

        return view;
    }

    public static class Contributors extends PreferenceFragmentCompat {

        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.contributors_list, rootKey);
        }
    }
}

File: app/src/main/java/com/drnoob/datamonitor/ui/fragments/ContributorsFragment.kt
/*
 * Copyright (C) 2021 Dr.NooB
 *
 * This file is a part of Data Monitor <https://github.com/itsdrnoob/DataMonitor>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.drnoob.datamonitor.ui.fragments

import android.content.Intent
import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat
import com.drnoob.datamonitor.R
import com.drnoob.datamonitor.core.Values.APP_CONTRIBUTORS_FRAGMENT
import com.drnoob.datamonitor.core.Values.GENERAL_FRAGMENT_ID
import com.drnoob.datamonitor.core.base.Preference
import com.drnoob.datamonitor.ui.activities.ContainerActivity
import com.drnoob.datamonitor.ui.activities.WallOfThanksActivity

class ContributorsFragment: PreferenceFragmentCompat() {
    private lateinit var appContributors: Preference
    private lateinit var wallOfThanks: Preference
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.contributors_preference, rootKey)

        appContributors = findPreference("app_contributors")!!
        wallOfThanks = findPreference("wall_of_thanks")!!

        appContributors.setOnPreferenceClickListener {
            context?.startActivity(Intent(context, ContainerActivity().javaClass)
                .putExtra(GENERAL_FRAGMENT_ID, APP_CONTRIBUTORS_FRAGMENT))
            false
        }

        wallOfThanks.setOnPreferenceClickListener {
            context?.startActivity(Intent(context, WallOfThanksActivity().javaClass))
            false
        }

    }
}